spring-boot-admin-client/src/main/java/de/codecentric/boot/admin/client/registration/RegistrationApplicationListener.java

```diff
@@ -94,4 +95,26 @@ public void setAutoRegister(boolean autoRegister) {
     public void setRegisterPeriod(Duration registerPeriod) {
         this.registerPeriod = registerPeriod;
     }
+
+    @Override
+    public void afterPropertiesSet() {
+        taskScheduler = registrationTaskScheduler();
+    }
+
+    private ThreadPoolTaskScheduler registrationTaskScheduler() {
+        final ThreadPoolTaskScheduler taskScheduler = new ThreadPoolTaskScheduler();
+        taskScheduler.setPoolSize(1);
+        taskScheduler.setRemoveOnCancelPolicy(true);
+        taskScheduler.setThreadNamePrefix("registrationTask");
+        taskScheduler.initialize();
+
+        return taskScheduler;
+    }
+
+    @Override
+    public void destroy() {
+        if (taskScheduler != null) {
```

Comment:
the null check isn't needed when having the taskScheduler final an being set in the constructor.